src/System.Net.Security/tests/FunctionalTests/KerberosTest.cs

```diff
@@ -0,0 +1,488 @@
+
+        public KerberosTest(KDCSetup fixture, ITestOutputHelper output)
+        {
+            _fixture = fixture;
+            _output = output;
+            _isKrbAvailable = _fixture.CheckAndClearCredentials(_output);
+        }
+
+        [Fact, OuterLoop]
+        [PlatformSpecific(PlatformID.Linux)]
+        public void NegotiateStream_StreamToStream_KerberosAuthentication_Success()
+        {
+            if (!_isKrbAvailable)
+            {
+                _output.WriteLine("skipping NegotiateStream_StreamToStream_KerberosAuthentication_Success");
+                return;
+            }
+
+            VirtualNetwork network = new VirtualNetwork();
+
+            using (var clientStream = new VirtualNetworkStream(network, isServer: false))
+            using (var serverStream = new VirtualNetworkStream(network, isServer: true))
+            using (var client = new NegotiateStream(clientStream))
+            using (var server = new UnixGssFakeNegotiateStream(serverStream))
+            {
+                Assert.False(client.IsAuthenticated, "client is not authenticated");
+
+                Task[] auth = new Task[2];
+                string user = string.Format("{0}@{1}", TestConfiguration.KerberosUser, TestConfiguration.Realm);
+                string target = string.Format("{0}@{1}", TestConfiguration.HostTarget, TestConfiguration.Realm);
+                NetworkCredential credential = new NetworkCredential(user, _fixture.password);
+                auth[0] = client.AuthenticateAsClientAsync(credential, target);
+                auth[1] = server.AuthenticateAsServerAsync();
+
+                bool finished = Task.WaitAll(auth, TestConfiguration.PassingTestTimeoutMilliseconds);
+                Assert.True(finished, "Handshake completed in the allotted time");
+
+                // Expected Client property values:
+                Assert.True(client.IsAuthenticated, "client.IsAuthenticated");
+                Assert.Equal(TokenImpersonationLevel.Identification, client.ImpersonationLevel);
+                Assert.True(client.IsEncrypted, "client.IsEncrypted");
+                Assert.True(client.IsMutuallyAuthenticated, "client.IsMutuallyAuthenticated");
+                Assert.False(client.IsServer, "client.IsServer");
+                Assert.True(client.IsSigned, "client.IsSigned");
+                Assert.False(client.LeaveInnerStreamOpen, "client.LeaveInnerStreamOpen");
+
+                IIdentity serverIdentity = client.RemoteIdentity;
+                Assert.Equal("Kerberos", serverIdentity.AuthenticationType);
+                Assert.True(serverIdentity.IsAuthenticated, "serverIdentity.IsAuthenticated");
+                IdentityValidator.AssertHasName(serverIdentity, target);
+            }
+        }
+
+        [Fact, OuterLoop]
+        [PlatformSpecific(PlatformID.Linux)]
+        public void NegotiateStream_StreamToStream_AuthToHttpTarget_Success()
+        {
+            if (!_isKrbAvailable)
```

Comment:
There are multiple constraints here - 
1. we need to invoke the fixture and install the pre-reqs, but only if the developer chooses to do so . 
2. If the developer does not choose to install and configure KDC (presumably because he/she is not testing negotiatestream, the outerloop should not fail). 

3. If the tests install anything, we need to uninstall them as well. 


Taken together, these three orthogonal requirements make it tricky to have it as a conditionalFact. 
Especially when we consider that xunit can do the filtering before or after the fixture is created. And we are relying the fixture's disposal to be called for clean up.
But I will track this  via #7031 (this is item 7)